packages/CodingStandard/src/Fixer/DependencyInjection/InjectToConstructorInjectionFixer.php

```diff
@@ -0,0 +1,239 @@
+
+use Nette\PhpGenerator\Method;
+use Nette\Utils\Strings;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\Fixer\DefinedFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+final class InjectToConstructorInjectionFixer implements DefinedFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Constructor injection should be used instead of @inject annotations.',
+            [
+                new CodeSample(
+'<?php
+class SomeClass
+{
+    /**
+     * @inject
+     * @var stdClass
+     */
+    public $property;
+}'),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_DOC_COMMENT, T_PUBLIC, T_VARIABLE]);
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        // 1. find annotation @inject
+        foreach ($tokens as $index => $token) {
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $injectAnnotations = $doc->getAnnotationsOfType('inject');
+            if (! count($injectAnnotations)) {
+                continue;
+            }
+
+            // 1. remove it
+            foreach ($injectAnnotations as $injectAnnotation) {
+                $injectAnnotation->remove();
+            }
+
+            $token->setContent($doc->getContent());
+
+            // 2. make public property private
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_PUBLIC)) {
+                    $token->override([T_PRIVATE, 'private']);
+                    break;
+                }
+            }
+
+            // 3. add dependency to constructor
+            $propertyName = '';
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_VARIABLE)) {
+                    $propertyName = ltrim($token->getContent(), '$');
+                    break;
+                }
+            }
+            $varAnnotation = $doc->getAnnotationsOfType('var')[0];
+            $propertyType = $varAnnotation->getTypes()[0];
+
+            // detect constructor
+            $constructMethodPosition = null;
+            for ($i = $index; $i < count($tokens); ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_FUNCTION)) {
+                    $namePosition = $tokens->getNextNonWhitespace($i);
+                    $methodNameToken = $tokens[$namePosition];
+                    if ($methodNameToken->getContent() === '__construct') {
```

Comment:
digging in my abandoned PR's I found this;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2526/files
which might help to find just those magic methods ;)